### PR TITLE
fix(Pagination): start converting to new css principles

### PR DESCRIPTION
**Pull request by eirikbacker**
- Part of #2295 
- Wraps pagination for now, but should be looked more into #2396
- Skipping #2283 for now to avoid too large PR
- Renames `Pagination.Content` to `Pagination.List` following #2221
- Fixes so `Button` with `hidden` attribute is actually hidden ☺️

### CODE REVIEW

File: packages/react/src/components/Pagination/PaginationList.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    const { size } = useContext(PaginationContext);
+    const Component = asChild ? Slot : 'ul';
+
+    return (
+      <Component
+        className={cl('ds-pagination', className)}
+        data-size={size}
+        ref={ref}
+        {...rest}
+      />
```
Comment (Barsnes) [inline packages/react/src/components/Pagination/PaginationList.tsx:26]: Since we are adding a default `aria-label` on `Pagination.tsx`, we can probably do it here as well.
Let's leave that for another PR, or do it here and create a separate changeset for it.

File: packages/react/src/components/Pagination/PaginationRoot.tsx [hunk 2]
```diff
@@ -14,6 +14,10 @@ export const PaginationContext = createContext<PaginationContextProps>({
 });
 
 export type PaginationRootProps = {
+  /**
+   * Sets screen reader label for the pagination area
+   */
+  'aria-label': string;
```
Comment (Barsnes) [inline packages/react/src/components/Pagination/PaginationRoot.tsx:20]: Lets do a default instead

Comment (Barsnes) [inline packages/react/src/components/Pagination/PaginationRoot.tsx:20]:
```suggestion
```
Lets do a default instead

File: packages/react/src/components/Pagination/PaginationList.tsx (hunk 1 above)
Comment (eirikbacker) [reply packages/react/src/components/Pagination/PaginationList.tsx:26]: Hi! Unsure what you mean - we should not have more than one wrapping aria-label, and that should be on the `<nav>`-element ☺️

File: packages/react/src/components/Pagination/PaginationRoot.tsx (hunk 2 above)
Comment (eirikbacker) [reply packages/react/src/components/Pagination/PaginationRoot.tsx:20]: Ah! Good catch, did not remember this was what we concluded in Breadcrumbs, I'll fix <3

**Comment by mimarz** [comment]
> * Part of [Preparations for CSS-package #2295](https://github.com/digdir/designsystemet/issues/2295)
> * Wraps pagination for now, but should be looked more into [Solution for Pagination wrapping/zoom #2396](https://github.com/digdir/designsystemet/issues/2396)
> * Does not fix [Streamline how to use `Pagination` in code #2283](https://github.com/digdir/designsystemet/issues/2283) to avoid too large PR
> * Renames `Pagination.Content` to `Pagination.List` following [Consistent component api #2221](https://github.com/digdir/designsystemet/issues/2221)
> * Fixes so `Button` with `hidden` attribute is actually hidden ☺️

Make sure to edit your third bullet point and remove "fix" infront of the issue as github keyword is a quite simple being 🐴

image

File: packages/react/src/components/Pagination/PaginationRoot.tsx [hunk 3]
```diff
@@ -14,6 +14,11 @@ export const PaginationContext = createContext<PaginationContextProps>({
 });
 
 export type PaginationRootProps = {
+  /**
+   * Sets the screen reader label for the Pagination area
+   * @default 'Sidenavigering'
+   */
+  'aria-label'?: string;
```
Comment (mimarz) [inline packages/react/src/components/Pagination/PaginationRoot.tsx:21]: Was this supposed to be required? ( according to your changeset 😅 )

Comment (eirikbacker) [reply packages/react/src/components/Pagination/PaginationRoot.tsx:21]: Oh, Tobias said this should not be required after all, but I forgot to remove this from changeset- well spotted! I'll update :)

**Comment by eirikbacker** [comment]
> > * Part of [Preparations for CSS-package #2295](https://github.com/digdir/designsystemet/issues/2295)
> > * Wraps pagination for now, but should be looked more into [Solution for Pagination wrapping/zoom #2396](https://github.com/digdir/designsystemet/issues/2396)
> > * Does not fix [Streamline how to use `Pagination` in code #2283](https://github.com/digdir/designsystemet/issues/2283) to avoid too large PR
> > * Renames `Pagination.Content` to `Pagination.List` following [Consistent component api #2221](https://github.com/digdir/designsystemet/issues/2221)
> > * Fixes so `Button` with `hidden` attribute is actually hidden ☺️
> 
> Make sure to edit your third bullet point and remove "fix" infront of the issue as github keyword is a quite simple being 🐴
> 
> image

Haha, nice :D

**Review by mimarz — APPROVED** [review]
